arbitrage_opportunity/detector.py

```diff
@@ -85,6 +85,46 @@ def get_best_opportunity(tickers: List[ShortTicker]) -> Tuple[List[ShortTicker],
 
     return best_triplet, best_profit
 
+def get_best_opportunity(tickers: List[ShortTicker]) -> Tuple[List[ShortTicker], float]:
+    # Build a directed graph of currencies
+    graph = nx.DiGraph()
+
+    for ticker in tickers:
+        if ticker.symbol is not None:
+            graph.add_edge(ticker.symbol.base, ticker.symbol.quote, ticker=ticker)
+            graph.add_edge(ticker.symbol.quote, ticker.symbol.base,
+                           ticker=ShortTicker(symbols.Symbol(f"{ticker.symbol.quote}/{ticker.symbol.base}"),
+                                              1 / ticker.last_price, reversed=True))
+
+    best_profit = 0
+    best_cycle = None
+
+    # Find all cycles in the graph (not just len = 3)
+    for cycle in nx.simple_cycles(graph):
```

Comment:
Keep in mind the new cycle detection code does not increase complexity (purely based on implementation) compared to the triangular detection code.

Reply:
I see, thanks for these details